src/program/latex/generation.rs

```diff
@@ -0,0 +1,7 @@
+use self::diagram::tikz::Tikz;
+
+pub mod diagram;
+
+pub trait LatexFactory {
+    fn create_tikz_diagram(&self) -> Box<dyn Tikz>;
```

Comment:
You probably want to use associated types instead of boxed trait objects:

```suggestion
    type Tikz: Tikz;
    fn create_tikz_diagram(&self) -> Self::Tikz;
```